### PR TITLE
CA-43 Initial networking tests

### PR DESCRIPTION
**Pull request by hbarros-caylent**
- Adds the folder `test` containing Terratest Golang code.
- Adds the folder `test_examples` for having testable TF root modules.
- Uses Terratest to assure the module gives out the outputs needed.

### CODE REVIEW

File: examples/minimal/variables.tf [hunk 1]
```diff
@@ -1,8 +1,34 @@
+variable "vpc_cidr_block" {
```
Comment (franco-caylent) [inline examples/minimal/variables.tf:1]: We should add descriptions and maybe default values for the new variables

File: test/test_util.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		}
+	})
+
+	t.Run("get_vpc", func(t *testing.T) {
+		// checks VPC exists by calling DescribeVPC
+		_, err := aws.GetVpcByIdE(t, outputs["vpc_id"].(string), awsRegion)
+		require.NoError(t, err, "Error trying to Describe VPC")
+	})
+
+	t.Run("check_vpc_name", func(t *testing.T) {
```
Comment (franco-caylent) [inline test/test_util.go:38]: Is this necessary here?

File: test/test_util.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			assert.Contains(t, allSubnetsOutput, s.Id)
+		}
+	})
+
+	t.Run("check_subnets_azs", func(t *testing.T) {
+		for _, subnet := range subnets {
+			assert.Contains(t, outputs["tamr_ec2_availability_zone"], subnet.AvailabilityZone)
+		}
+	})
+}
```
Comment (franco-caylent) [inline test/test_util.go:58]: Nice checks!

File: test_examples/minimal/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Public Facing Example
```
Comment (franco-caylent) [inline test_examples/minimal/README.md:1]: You are going to need to run make terraform docs to update this

**Review by franco-caylent — COMMENTED** [review]
In good shape!

File: examples/minimal/variables.tf (hunk 1 above)
Comment (hbarros-caylent) [reply examples/minimal/variables.tf:1]: done in https://github.com/Datatamer/terraform-aws-networking/pull/5/commits/ab73ab276d2dca3f30b38b215e9efd74943974e8

File: test_examples/minimal/README.md (hunk 4 above)
Comment (hbarros-caylent) [reply test_examples/minimal/README.md:1]: done https://github.com/Datatamer/terraform-aws-networking/pull/5/commits/8a1a924a58aaf980ab3b049af7778eeff6216143

File: examples/minimal/vpc.tf [hunk 5]
```diff
@@ -1,15 +1,15 @@
 module "tamr_networking" {
   #source = "git::https://github.com/Datatamer/terraform-aws-networking.git?ref=0.1.0"
   source                        = "../../"
-  ingress_cidr_blocks           = ["172.16.0.0/16"]
-  vpc_cidr_block                = "10.0.0.0/16"
-  data_subnet_cidr_blocks       = ["10.0.2.0/24", "10.0.3.0/24"]
-  application_subnet_cidr_block = "10.0.0.0/24"
-  compute_subnet_cidr_block     = "10.0.1.0/24"
+  ingress_cidr_blocks           = var.ingress_cidr_blocks           # ["172.16.0.0/16"]
```
Comment (souza-dan) [inline examples/minimal/vpc.tf:4]: nit: Can cleanup those commented values.

File: test_examples/minimal/README.md (hunk 4 above)
Comment (souza-dan) [inline test_examples/minimal/README.md:1]: nit: this header is a bit misleading. More of a test case than example. Could probably just remove it unless there's something meaningful to be said here.

**Review by souza-dan — APPROVED** [review]
LGTM, just a few super nit-picky comments from me.

**Pull request merged**